stable/database/templates/_helpers.tpl

```diff
@@ -247,6 +247,23 @@ The configuration is imported only in the entrypoint cluster.
 {{- end -}}
 {{- end -}}
 
+{{/*
+Enables automatic database protocol upgrade via annotations.
+The configuration is imported only in the entrypoint cluster.
+*/}}
+{{- define "database.automaticProtocolUpgrade" -}}
+{{- if .Values.database.automaticProtocolUpgrade }}
+{{- if eq (include "defaultfalse" .Values.database.automaticProtocolUpgrade.enabled) "true" -}}
+{{- if (eq (default "cluster0" .Values.cloud.cluster.name) (default "cluster0" .Values.cloud.cluster.entrypointName)) }}
+"nuodb.com/automatic-database-protocol-upgrade": "true"
+{{- with .Values.database.automaticProtocolUpgrade.tePreferenceQuery }}
+"nuodb.com/automatic-database-protocol-upgrade.te-preference-policy": {{ . | quote }}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
```

Comment:
I don't know if Helm allows it, but if it does I think it would be better to use indentation on nested these control structures to make them readable, e.g.

```
{{- define "database.automaticProtocolUpgrade" -}}
  {{- if .Values.database.automaticProtocolUpgrade }}
    {{- if eq (include "defaultfalse" .Values.database.automaticProtocolUpgrade.enabled) "true" -}}
      {{- if (eq (default "cluster0" .Values.cloud.cluster.name) (default "cluster0" .Values.cloud.cluster.entrypointName)) }}
      "nuodb.com/automatic-database-protocol-upgrade": "true"
        {{- with .Values.database.automaticProtocolUpgrade.tePreferenceQuery }}
        "nuodb.com/automatic-database-protocol-upgrade.te-preference-policy": {{ . | quote }}
        {{- end -}}
      {{- end -}}
    {{- end -}}
  {{- end -}}
{{- end -}}
```

Reply:
Yes, we can nest the control structures.
We should just make sure that the indentation for the rendered text is correct.  For example, the actual annotation should be with 0 indentation because the whole template is indented with 4 spaces later when called.
I will make the change.